File: fsharp/lucians-luscious-lasagna/LuciansLusciousLasagna.fs
module LuciansLusciousLasagna

// TODO: define the 'expectedMinutesInOven' binding
let expectedMinutesInOven: int = 40

// TODO: define the 'remainingMinutesInOven' function
let remainingMinutesInOven (minutes: int): int = expectedMinutesInOven - minutes

// TODO: define the 'preparationTimeInMinutes' function
let preparationTimeInMinutes (layers: int): int = layers * 2

// TODO: define the 'elapsedTimeInMinutes' function
let elapsedTimeInMinutes (layers: int) (minutes: int): int = preparationTimeInMinutes layers + minutes

File: csharp/lucians-luscious-lasagna/LuciansLusciousLasagna.cs
class Lasagna
{
    public int ExpectedMinutesInOven()
    {
        return 40;
    }

    public int RemainingMinutesInOven(int passed)
    {
        return ExpectedMinutesInOven() - passed;
    }

    public int PreparationTimeInMinutes(int layers)
    {
        return layers * 2;
    }

    public int ElapsedTimeInMinutes(int layers, int passed)
    {
        return PreparationTimeInMinutes(layers) + passed;
    }
}
